disnake/poll.py

```diff
@@ -0,0 +1,442 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from . import utils
+from .abc import Snowflake
+from .emoji import Emoji, _EmojiTag
+from .enums import PollLayoutType, try_enum
+from .iterators import PollAnswerIterator
+from .partial_emoji import PartialEmoji
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from .abc import MessageableChannel
+    from .message import Message
+    from .state import ConnectionState
+    from .types.poll import (
+        Poll as PollPayload,
+        PollAnswer as PollAnswerPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollCreateAnswerPayload,
+        PollCreateMediaPayload,
+        PollCreatePayload,
+        PollMedia as PollMediaPayload,
+        PollResult as PollResultPayload,
+    )
+
+__all__ = (
+    "PollResult",
+    "PollAnswerCount",
+    "PollMedia",
+    "PollAnswer",
+    "Poll",
+)
+
+
+class PollAnswerCount:
+    """Represents a poll answer count from discord.
+
+    .. versionadded:: 2.10
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The ID of the answer the counts are of.
+    count: :class:`int`
+        The number of votes for this answer.
+    me_voted: :class:`bool`
+        Whether the current user voted for this answer.
+    """
+
+    __slots__ = (
+        "id",
+        "count",
+        "me_voted",
+    )
+
+    def __init__(self, data: PollAnswerCountPayload) -> None:
+        self.id: int = int(data["id"])
+        self.count: int = int(data["count"])
+        self.me_voted: bool = data["me_voted"]
+
+    def __repr__(self) -> str:
+        return (
+            f"<{self.__class__.__name__} id={self.id} count={self.count} me_voted={self.me_voted}>"
+        )
+
+
+class PollResult:
+    """Represents a poll result from discord.
+
+    .. versionadded:: 2.10
+
+    Attributes
+    ----------
+    is_finalized: :class:`bool`
+        Whether the votes have been precisely counted.
+    answer_counts: List[:class:`PollAnswerCount`]
+        The counts for each answer.
+    """
+
+    __slots__ = (
+        "is_finalized",
+        "answer_counts",
+    )
+
+    def __init__(self, data: PollResultPayload) -> None:
+        self.is_finalized: bool = data["is_finalized"]
+        self.answer_counts: List[PollAnswerCount] = [
+            PollAnswerCount(answer) for answer in data["answer_counts"]
+        ]
```

Comment:
looking at this again, it makes me wonder if these two fields could be folded into `Poll`/`PollAnswer` respectively, so that `PollResult` and `PollAnswerCount` are removed entirely.
This would make it easier to update from the gateway event (see other comment), and simplifies retrieving an answer's vote count (`answer.vote_count` or similar, instead of `utils.get(poll.results.answer_counts, id=answer.id).count`)

Reply:
Yeah that seems fine, hoping that discord doesn't add other things to these models